src/lib/crypto/kimchi_backend/gadgets/range_check.ml

```diff
@@ -2,16 +2,23 @@ open Core_kernel
 
 open Kimchi_backend_common.Plonk_constraint_system.Plonk_constraint
 
-(* 64-bit range-check gadget *)
-let range_check64 (type f)
+(* Helper to create RangeCheck0 gate, configured in various ways
+ *     - is_64bit   : create 64-bit range check
+ *     - is_compact : compact limbs mode (only used by compact multi-range-check)
+ *)
+let range_check0 (type f)
     (module Circuit : Snarky_backendless.Snark_intf.Run with type field = f)
+    ?(is_64bit : bool = false) ?(is_compact : bool = false)
     (v0 : Circuit.Field.t) =
   let open Circuit in
-
   (* Define a shorthand helper *)
```

Comment:
ah so I think at this point, you should do something like this:

```ocaml
match Circuit.Field.to_constant v0 with
  | Some x -> (* do your range check on the constant directly, and return an error/assert if it's wrong *)
  | _ -> (* the rest of your code *)
```

and forget what I told you previously about doing this in that weird `eval` function :D

Reply:
Unless this is easy and quick to do now, I'm for opening an issue and adding a TODO. We can always revisit later.

Reply:
I've created a issue for this here https://github.com/MinaProtocol/mina/issues/12838